utility/src/main/java/org/oppia/android/util/locale/DisplayLocaleImpl.kt

```diff
@@ -13,16 +13,18 @@ import java.util.Locale
 import java.util.Objects
 
 // TODO(#3766): Restrict to be 'internal'.
-/** Implementation of [OppiaLocale.DisplayLocale]. */
+// TODO(#3766): Restrict formattingLocale to be 'internal'.
+/**
+ * Implementation of [OppiaLocale.DisplayLocale].
+ *
+ * @property formattingLocale the [Locale] used for user-facing string formatting
```

Comment:
Add other properties.

Also, explain why this change was needed in PR description (creating a locale is no longer synchronous).

Reply:
Updated PR description.

Reply:
Updated docs in latest changes.